### PR TITLE
openstack: Use external's network DNS by default

### PR DESCRIPTION
**Pull request by flaper87**

### CODE REVIEW

**Comment by wjiangjay** [comment]
This is for coredns right?

File: data/data/openstack/topology/private-network.tf [hunk 1]
```diff
@@ -3,18 +3,25 @@ locals {
   service_cidr_block = "${cidrsubnet(var.cidr_block, 1, 1)}"
 }
 
+data "openstack_networking_network_v2" "external_network" {
+  name       = "${var.external_network}"
```
Comment (wjiangjay) [inline data/data/openstack/topology/private-network.tf]: Do here need prefix?

Comment (flaper87) [reply data/data/openstack/topology/private-network.tf]: no, this is a query to ge the existing, shared, external network. It's not creating a resource.

Comment (wjiangjay) [reply data/data/openstack/topology/private-network.tf]: oh, got you

File: data/data/openstack/topology/private-network.tf [hunk 2]
```diff
@@ -3,18 +3,25 @@ locals {
...
-  cidr       = "${local.service_cidr_block}"
-  ip_version = 4
-  network_id = "${openstack_networking_network_v2.openshift-private.id}"
-  tags       = ["openshiftClusterID=${var.cluster_id}"]
+  name            = "${var.cluster_id}-service"
+  cidr            = "${local.service_cidr_block}"
+  ip_version      = 4
+  network_id      = "${openstack_networking_network_v2.openshift-private.id}"
+  tags            = ["openshiftClusterID=${var.cluster_id}"]
+  dns_nameservers = ["128.31.24.12", "128.31.24.11"]
```
Comment (wjiangjay) [inline data/data/openstack/topology/private-network.tf]: if the openstack is not public resolved, here should be an issue.

```
(openstack) console log show wjiang-ocp-qrcck-api 
[K[    [31m*[1;31m*[0m] A start job is running for Ignition (disks) (14s / no limit)[   17.252586] ignition[455]: GET https://rhos-d.infra.prod.upshift.rdu2.redhat.com:13808/v1/AUTH_542c6ebd48bf40fa857fc245c7572e30/wjiang-ocp.shiftstack.com/load-balancer.ign?temp_url_sig=b109572b95db65670a5a5f24ec77b0995ef428be&temp_url_expires=1551782155: attempt #4
[   17.469253] ignition[455]: GET error: Get https://rhos-d.infra.prod.upshift.rdu2.redhat.com:13808/v1/AUTH_542c6ebd48bf40fa857fc245c7572e30/wjiang-ocp.shiftstack.com/load-balancer.ign?temp_url_sig=b109572b95db65670a5a5f24ec77b0995ef428be&temp_url_expires=1551782155: dial tcp: lookup rhos-d.infra.prod.upshift.rdu2.redhat.com on 128.31.24.11:53: server misbehaving
```

Comment (flaper87) [reply data/data/openstack/topology/private-network.tf]: A public network is a hard requirement for the installer, IIRC. We must be able to pull containers from the registries so we can expect there to be an external network.

Comment (wjiangjay) [reply data/data/openstack/topology/private-network.tf]: But here, the loadbalancer is try to get its ignition to bootstrap, but failed due to opestack is an internal host which can not be resolved by 128.31.24.1{1,2}.

**Pull request merged**